src/engine/Match_tainting_mode.mli

```diff
@@ -31,7 +41,13 @@ val hook_setup_hook_function_taint_signature :
   *)
 
 (* It could be a private function, but it is also used by Deep Semgrep. *)
+(* This [formula_cache] argument is exposed here because this function is also
+   a subroutine but the cache itself should be created outside of the any main
+   loop which runs over rules. This cache is only safe to share with if
+   [taint_config_of_rule] is used on the same file!
+*)
 val taint_config_of_rule :
+  formula_cache:formula_cache ->
```

Comment:
Perhaps to make it even more clear we could name this parameter `per_file_formula_cache` ?